### ISSUE TITLE
SDO and PDO parallel execution

### ISSUE THREAD

**Issue by nkt17** [OpenEtherCATsociety/SOEM#867]
Good day everyone.
I need periodically request not mappable object by SDO. So the question occured. Can we parallel execute next actions?

1.
ec_send_processdata();
ec_receive_processdata(EC_TIMEOUTRET);

and
2.
ec_SDOread(1, index, subindex, FALSE, &size1, &v1, EC_TIMEOUTSAFE);

**Comment by nkt17** [comment]
Also when i'm trying to request object by SDO sometimes there is incorrect data
![image](https://github.com/user-attachments/assets/3ceb5eab-d1a9-4dbf-8845-7ecce5d2e9fd)

**Comment by ArthurKetels** [comment]
Yes, PDO and SDO can function perfectly next to each other. But make sure they are running in different threads. Also check the return value of any SDO function, they might fail. And when it fails check the error message that is stored in the error list.

**Comment by nkt17** [comment]
> Yes, PDO and SDO can function perfectly next to each other. But make sure they are running in different threads. Also check the return value of any SDO function, they might fail. And when it fails check the error message that is stored in the error list.

Good day. Thanks for response.
I removed lock_guard:
![image](https://github.com/user-attachments/assets/d7d69cd4-2cd4-4cdf-a38a-7dfe463fccd6)

I use this method to request parameters on the next photo (right values).
![image](https://github.com/user-attachments/assets/6230a4ab-4fb5-4442-9111-e15addd84576)

On removing lock_guard all the parameters became wrong (result on the next photo)
![image](https://github.com/user-attachments/assets/0a0291b4-358a-4eb6-ab66-e3bb741bba4e)

**Comment by ArthurKetels** [comment]
OK, but did you do it the way I instructed? You do not show any of your code or a wireshark trace.
